src/Application/EnvironmentVariables/Commands/UpdateEnvironmentVariablesCommand.cs

```diff
@@ -0,0 +1,83 @@
+using Hippo.Application.Common.Interfaces;
+using Hippo.Core.Entities;
+using MediatR;
+
+namespace Hippo.Application.EnvironmentVariables.Commands;
+
+public class UpdateEnvironmentVariablesCommand : IRequest
+{
+    public List<EnvironmentVariableRequest> EnvironmentVariables { get; set; } = new List<EnvironmentVariableRequest>();
+}
+
+public class UpdateEnvironmentVariablesCommandHandler : IRequestHandler<UpdateEnvironmentVariablesCommand>
+{
+    private readonly IApplicationDbContext _context;
+
+    public UpdateEnvironmentVariablesCommandHandler(IApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<Unit> Handle(UpdateEnvironmentVariablesCommand request, CancellationToken cancellationToken)
+    {
+        var existingVariables = GetExistingEnvironmentVariables(request.EnvironmentVariables.First().ChannelId);
+
+        AddNewEnvironmentVariables(request.EnvironmentVariables);
+        UpdateExistingEnvironmentVariables(existingVariables, request.EnvironmentVariables);
+        RemoveDeletedEnvironmentVariables(existingVariables, request.EnvironmentVariables);
+
+        await _context.SaveChangesAsync(cancellationToken);
+
+        return Unit.Value;
+    }
+
+    private List<EnvironmentVariable> GetExistingEnvironmentVariables(Guid channelId)
+    {
+        return _context.EnvironmentVariables
+            .Where(v => v.Id == channelId)
+            .ToList();
+    }
+
+    private void AddNewEnvironmentVariables(List<EnvironmentVariableRequest> environmentVariables)
```

Comment:
I'd prefer if this was a static method (pure method) that would not update the context directly. 
Same for the other 2 and the _context would be updated in a single method. The one that calls these 3 methods.